serde-support/src/main/java/io/micronaut/serde/support/util/SerdeArgumentConf.java

```diff
@@ -83,6 +84,22 @@ public SerdeArgumentConf(AnnotationMetadata annotationMetadata) {
         this.subtypeInfo = SubtypeInfo.createForProperty(annotationMetadata);
     }
 
+    /**
+     * Reconstruct the generic type with parent metadata if necessary.
+     * @param parentType The parent type
+     * @param valueGeneric The generic
+     * @return The new argument
+     * @param <V> The value generic
+     */
+    public static <V> @NotNull Argument<V> reconstructGenericWithParentMetadata(@NotNull Argument<?> parentType, @NotNull Argument<V> valueGeneric) {
+        return Argument.of(
```

Comment:
Can you use `valueGeneric.withAnnotationMetadata`? It might be in the newer version